djiffy/models.py

```diff
@@ -95,6 +97,62 @@ def rights_statement_id(self):
         if self.license and 'rightsstatements.org' in self.license:
             return self.license.rstrip(' /').split('/')[-2]
 
+    _rights_graph = None
+
+    def license_label(self, lang='en'):
+        '''Get the text label for the rights license.  Uses local
+        value from edm rights if available; otherwise uses
+        data for the URI to get the preferred label or title.'''
+
+        # Some manifests havea seeAlso data contains an "edm_rights"
+        # section with a label for the rights statement.
+        # Use that if available (NOTE: ignores specified language)
+        # NOTE: possibly PUL specific, but shouldn't hurt to look locally first
+        for data in self.extra_data.values():
+            if 'edm_rights' in data and 'pref_label' in data['edm_rights']:
+                return data['edm_rights']['pref_label']
+
+        # if license/rights label is not available locally, get via uri
+        if self._rights_graph is None:
+            # if license is defined and a url
+            if self.license and urllib.parse.urlparse(self.license).scheme in ['http', 'https']:
+                self._rights_graph = rdflib.Graph()
+                try:
+                    # rights statement org does content-negotiation for json-jd,
+                    # but rdflib doesn't handle that automatically
+                    if 'rightsstatements.org' in self.license:
+                        resp = requests.get(self.license,
+                                            headers={'Accept': 'application/json'},
+                                            allow_redirects=False)
+                        if resp.status_code == requests.codes.see_other:
+                            self._rights_graph.parse(resp.headers['location'], format='json-ld')
+
+                    # creative commons doesn't support content negotiation,
+                    # but you can add rdf to the end of the url
+                    elif 'creativecommons.org' in self.license:
+                        rdf_uri = '/'.join([self.license.rstrip('/'), 'rdf'])
+                        self._rights_graph.parse(rdf_uri)
+
+                except Exception:
```

Comment:
(Github ate my comment.)

I think this can use from `rdflib.exceptions.Error` and `requests.exception.RequestException` as those should encompass all the errors we just want to pass, and therefore avoid a bare `Exception`.